src/main/scala/io/iohk/ethereum/blockchain/sync/SyncStateSchedulerActor.scala

```diff
@@ -97,73 +167,298 @@ class SyncStateSchedulerActor(downloader: ActorRef, sync: SyncStateScheduler, sy
     }
   }
 
-  // scalastyle:off method.length
+  private def processNodes(
+      currentState: SchedulerState,
+      currentStats: ProcessingStatistics,
+      currentDownloaderState: DownloaderState,
+      requestResult: RequestResult
+  ): Future[ProcessingResult] = {
```

Comment:
Does it make sense then to write this function without `Future` and wrap into Future at call site?